app/app_test.go

```diff
@@ -64,11 +65,11 @@ func TestInitChain(t *testing.T) {
 	traceStore := &NoopWriter{}
 	timeoutCommit := time.Second
 	appOptions := NoopAppOptions{}
-	testApp := app.New(logger, db, traceStore, timeoutCommit, appOptions)
+	testApp := app.New(logger, db, traceStore, timeoutCommit, appOptions, baseapp.SetChainID(testfactory.ChainID))
```

Comment:
genesis chainID and app chainID must batch

Reply:
indeed, they must... 

this testfactory thing seems a bit magic to me 🤔